### PR TITLE
GAP-2070 delete applicant

### PR DESCRIPTION
**Pull request by dominicwest**
Hard deletes an applicant after an admin deletes them. Yeets their entries from the following tables:
- Submissions
- Applicant org details
- grant export
- attachements

Two known caveats:
- Preserving data in grant_beneficiary as its not personally identifiably & useful for stats
- If an admin has already downloaded a deleted users submitted submissions, they will still of course have that data

### CODE REVIEW

File: src/main/java/gov/cabinetofice/gapuserservice/service/user/OneLoginUserService.java [hunk 1]
```diff
@@ -161,10 +164,17 @@ private boolean doesUserHaveDepartment(User user) {
...
-        return user;
+    @Transactional
+    public void deleteUser(Integer id, String jwt) {
+        final User user = userRepository.findById(id).orElseThrow(() -> new UserNotFoundException("user with id: " + id + "not found"));
+        webClientBuilder.build()
+                .delete()
+                .uri(adminBackend + "/users/delete/" + (user.hasSub() ? user.getSub() : "") + (user.hasColaSub() ? "?colaSub=" + user.getColaSub() : ""))
+                .header("Authorization", "Bearer " + jwt)
+                .retrieve()
+                .bodyToMono(Void.class)
```
Comment (dylanwrightCO) [inline src/main/java/gov/cabinetofice/gapuserservice/service/user/OneLoginUserService.java:175]: Thinking we should check the response status and cancel the transaction if unsuccessful?

File: src/main/java/gov/cabinetofice/gapuserservice/util/HelperUtils.java [hunk 2]
```diff
@@ -82,4 +86,11 @@ public static String generateSecureRandomString(final Integer strLen) {
     public static String generateUUID() {
         return UUID.randomUUID().toString();
     }
+
+    public static Cookie getCustomJwtCookieFromRequest(final HttpServletRequest request, final String userServiceCookieName) {
+        final Cookie customJWTCookie = WebUtils.getCookie(request, userServiceCookieName);
+        if (customJWTCookie == null)
+            throw new UnauthorizedException(userServiceCookieName + " cookie not found");
```
Comment (dylanwrightCO) [inline src/main/java/gov/cabinetofice/gapuserservice/util/HelperUtils.java:93]: Can add a test here to cover if cookie is null

File: src/main/java/gov/cabinetofice/gapuserservice/service/user/OneLoginUserService.java (hunk 1 above)
Comment (westdom) [reply src/main/java/gov/cabinetofice/gapuserservice/service/user/OneLoginUserService.java:175]: bodyToMono will throw a `WebClientException` if there's a 400 or 500 code so that should be fine

**Review by dylanwrightCO — APPROVED** [review]
